import/src/schema/validate.rs

```diff
@@ -0,0 +1,133 @@
+use crate::AggregateTSMSchema;
+use thiserror::Error;
+
+// Possible validation errors
+#[derive(Debug, Error)]
+pub enum ValidationError {
+    #[error("Measurement '{measurement}' has a tag and field with the same name: {name}")]
+    TagAndFieldSameName { measurement: String, name: String },
+
+    #[error(
+        "Measurement '{measurement}' has field '{name}' with multiple types: {:?}",
+        types
+    )]
+    FieldWithMultipleTypes {
+        measurement: String,
+        name: String,
+        types: Vec<String>,
+    },
+}
+
+pub fn validate_schema(schema: &AggregateTSMSchema) -> Result<(), Vec<ValidationError>> {
+    let mut errors: Vec<ValidationError> = vec![];
+    for measurement in &schema.measurements {
```

Comment:
It might be easier to read if this was something like:

```suggestion
    for (measurement_name, measurement) in &schema.measurements {
```

And then you could use `measurement_name` in the loop body rather than `measurement.0` and `measurement` instead of `measurement.1`

Reply:
> You also consolidated some of code out of schema_parse too, right?

i did, but the hakari bot's subsequent commit relegated by helpful comment to the dustbin of history :cry: 

this code in `lib.rs` you're commenting on is the move out of `schema_parse.rs`

Reply:
took your suggestion, thanks @alamb